mina.netty/src/main/java/org/kaazing/mina/filter/codec/statemachine/ConsumeToDynamicTerminatorDecodingState.java

```diff
@@ -110,10 +110,25 @@ public DecodingState decode(IoBuffer in, ProtocolDecoderOutput out)
             buffer.setAutoExpander(allocator);
         }
         buffer.put((IoBufferEx) in);
+        validate(buffer, beginPos);
         return this;
     }
 
     /**
+     * Validate partially received message, from {@code startPos} to the buffer's limit, in
+     * order to fail fast if what was received does not comply with what the current state
+     * is waiting for.
+     * 
+     * @param buffer the buffer containing the partial message 
+     * @param startPos the starting position inside the buffer
+     * @throws Exception if the validation fails or if any problems are encountered in the
+     *         validation process
+     */
+    protected void validate(IoBufferEx buffer, int startPos) throws Exception {
```

Comment:
@jitsni If you decide not to fix it, I can change the script to send something else other than "Hello", so it passes. Let me know how you want to proceed. Thanks!